src/vcpkg/registries.cpp

```diff
@@ -481,7 +481,7 @@ namespace
             return (*pscf)->to_version();
         }
         print_error_message(maybe_scf.error());
```

Comment:
Should this even print something to the user? 
The function return is `Optional<Version>` so not finding something should never be an error.

Reply:
The trouble with this particular class and this particular function is the scenario of "User edits ports/foo/vcpkg.json and introduces a typo." We need to ensure that the various error messages produced throughout the tool in that configuration are not misleading.

I think this should follow the same pattern as BuiltinFilesRegistry::get_port_entry -- if the folder exists, then it _must_ load or it is a critical failure (print the error and terminate). If the folder doesn't exist, then return nullopt.

Reply:
> is the scenario of "User edits ports/foo/vcpkg.json and introduces a typo."

Is that really a scenario? Aren't we in the `already installed` case here where the version doesn't matter any longer. Be aware that in manifest mode a change in `vcpkg.json` will trigger an automatic removal of said port (since the hash changed). So the only case we need to consider is the classic case of `vcpkg install` which does not support versioning.

> I think this should follow the same pattern as BuiltinFilesRegistry::get_port_entry -- if the folder exists, then it must load or it is a critical failure (print the error and terminate). If the folder doesn't exist, then return nullopt.

I don't think that lies in the responsibility of this function. 

The signature is:
`Optional<Version> BuiltinFilesRegistry::get_baseline_version(StringView port_name) const`
Why should this error if the folder for the port exists but doesn't load?


The signature for `BuiltinFilesRegistry::get_port_entry` is:
`std::unique_ptr<RegistryEntry> BuiltinFilesRegistry::get_port_entry(StringView port_name) const`
and it also doesn't error if the port folder doesn't load. It only errors if the directory name and the name in the manifest doesn't agree, e.g. the check is: `scf->core_paragraph->name == port_name`

It could be rewritten to:
```
      if (!m_fs.exists(port_directory, IgnoreErrors{})) {
          return nullptr; }

      auto scfp= get_scf(port_directory).get();

      if (!scfp) {
          return nullptr; }

      auto& scf = *scfp;
      if (!scf->core_paragraph->name == port_name)
      {
                Checks::exit_maybe_upgrade(VCPKG_LINE_INFO,
                                         "Error: Failed to load port from %s: names did not match: '%s' != '%s'",
                                         port_directory,
                                         port_name,
                                         scf->core_paragraph->name);

      }
      return std::make_unique<BuiltinPortTreeRegistryEntry>(
                    scf->core_paragraph->name, port_directory, scf->to_version());
```